### PR TITLE
SOLR-14623: Minor Improvements to SolrCloud tests

### PR DESCRIPTION
**Pull request by chatman**
Here are some of the changes herein:
1. MiniSolrCloudCluster: Changing the polling to waitForState (based on watch/latch)
2. Improve waiting for Jetty stop, wait until node is not in liveNodes (JettySolrRunner)
3. TestCloudConsistency: Explicitly pick the jetty hosting shard1's leader
4. TestWaitForStateWithJettyShutdowns: Wait until replica becomes active
5. TestCollectionsAPIViaSolrCloudCluster: Don't wait for the collection to be deleted fully at the end
6. TestRecovery: Unset the UpdateLog hooks

### CODE REVIEW

File: solr/core/src/test/org/apache/solr/cloud/TestCloudConsistency.java [hunk 1]
```diff
@@ -207,18 +208,38 @@ private void addDocToWhenOtherReplicasAreDown(String collection, Replica leader,
...
    */
   private void addDocWhenOtherReplicasAreNetworkPartitioned(String collection, Replica leader, int docId) throws Exception {
-    for (int i = 0; i < 3; i++) {
-      proxies.get(cluster.getJettySolrRunner(i)).close();
+    DocCollection col = cluster.getSolrClient().getZkStateReader().getClusterState().getCollection(collection);
+    Replica shard1Leader = col.getLeader("shard1");
+    String baseUrl = shard1Leader.getBaseUrl();
+    JettySolrRunner j1 = null; // This will be the jetty hosting shard1's leader
+    for (JettySolrRunner j : cluster.getJettySolrRunners()) {
+      System.out.println("cmp:" + j.getProxyBaseUrl() + " " + baseUrl);
```
Comment (tflobbe) [inline solr/core/src/test/org/apache/solr/cloud/TestCloudConsistency.java:216]: Is this intentionally left here?

File: solr/core/src/test/org/apache/solr/cloud/TestCloudConsistency.java [hunk 2]
```diff
@@ -207,18 +208,38 @@ private void addDocToWhenOtherReplicasAreDown(String collection, Replica leader,
...
+      if (j.getProxyBaseUrl().toString().equals(baseUrl)) {
+        j1 = j;
+        break;
+      }
+    }
+
+    assertNotNull(baseUrl, j1);
```
Comment (tflobbe) [inline solr/core/src/test/org/apache/solr/cloud/TestCloudConsistency.java:223]: Maybe you can improve this message? A failure would look confusing

File: solr/core/src/test/org/apache/solr/cloud/TestCloudConsistency.java [hunk 3]
```diff
@@ -207,18 +208,38 @@ private void addDocToWhenOtherReplicasAreDown(String collection, Replica leader,
...
-      proxies.get(cluster.getJettySolrRunner(i)).reopen();
+    for (JettySolrRunner j : cluster.getJettySolrRunners()) {
+      if (j != j1) {
+        proxies.get(j).reopen();
+      }
     }
     waitForState("Timeout waiting for leader goes DOWN", collection, (liveNodes, collectionState)
-        -> collectionState.getReplica(leader.getName()).getState() == Replica.State.DOWN);
+        ->  collectionState.getReplica(leader.getName()).getState() == Replica.State.DOWN);
+    Thread.sleep(1000);
```
Comment (tflobbe) [inline solr/core/src/test/org/apache/solr/cloud/TestCloudConsistency.java:242]: Is this sleep here to wait for the stop? Why is it now needed?

File: solr/core/src/test/org/apache/solr/cloud/TestCloudConsistency.java [hunk 4]
```diff
@@ -207,18 +208,38 @@ private void addDocToWhenOtherReplicasAreDown(String collection, Replica leader,
...
 
     // the meat of the test -- wait to see if a different replica become a leader
     // the correct behavior is that this should time out, if it succeeds we have a problem...
```
Comment (tflobbe) [inline solr/core/src/test/org/apache/solr/cloud/TestCloudConsistency.java:245]: Suggestion, if the expected behavior is the timeout here (no leader elected), WDYT about waiting for less time in normal tests (1second maybe?) and only doing 10 seconds on nightly?

File: solr/core/src/test/org/apache/solr/cloud/TestCloudConsistency.java [hunk 5]
```diff
@@ -207,18 +208,38 @@ private void addDocToWhenOtherReplicasAreDown(String collection, Replica leader,
    * Leader should be on node - 0
```
Comment (tflobbe) [inline solr/core/src/test/org/apache/solr/cloud/TestCloudConsistency.java:208]: I guess this is no longer true?

File: solr/core/src/test/org/apache/solr/cloud/TestCloudConsistency.java [hunk 6]
```diff
@@ -207,18 +208,38 @@ private void addDocToWhenOtherReplicasAreDown(String collection, Replica leader,
...
    */
   private void addDocWhenOtherReplicasAreNetworkPartitioned(String collection, Replica leader, int docId) throws Exception {
-    for (int i = 0; i < 3; i++) {
-      proxies.get(cluster.getJettySolrRunner(i)).close();
+    DocCollection col = cluster.getSolrClient().getZkStateReader().getClusterState().getCollection(collection);
+    Replica shard1Leader = col.getLeader("shard1");
+    String baseUrl = shard1Leader.getBaseUrl();
+    JettySolrRunner j1 = null; // This will be the jetty hosting shard1's leader
```
Comment (tflobbe) [inline solr/core/src/test/org/apache/solr/cloud/TestCloudConsistency.java:214]: Maybe `s/j1/leaderJetty/` or something?

**Comment by chatman** [comment]
Closing this, all of these improvements are covered in SOLR-14636.

**Pull request closed without merging**